arangod/IResearch/IResearchDataStore.cpp

```diff
@@ -1684,6 +1684,7 @@ void IResearchDataStore::truncateCommit(TruncateGuard&& guard,
   };
 
   TRI_IF_FAILURE("ArangoSearchTruncateFailure") {
+    CrashHandler::setHardKill();
```

Comment:
I don't sure. I'm ok with such fix.

But in general the most important question here is
How process should exit in case of some exception was thrown after truncate written to rocksdb wal,
but before truncate applied to all indexes.

Is it ok terminate in such case? For users, not for tests!
I asked that last time and there are no concerns about that.
But I think I need to explicitly ask it again.

Reply:
should that question rather be in a BTS since its not the main topic of the current issue?